### PR TITLE
Add/Edit Products: edit Tax status and Tax class in price settings

### PR DESCRIPTION
**Pull request by pmusolino**
Part of #1423

## Description
This PR implements inside the Edit Price Settings Screen all the logic behind Tax Status and Tax Class.
Part of another PR:
- Remote update of the price

## Changes
- Added a computed property `productTaxStatus` in the `Product` model.
- New model `ProductTaxStatus`
- Managed tax class and tax status inside the ProductPrice Settings View Controller, which present the two view controller pickers.
- Implemented `ProductTaxClassStatusListSelectorDataSource` + tests
- Implemented `ProductTaxClassListSelectorDataSource` + tests
- Introduced `isSelected()` method protocol in `PaginatedListSelectorViewController`
- Introduced `isSelected()` method protocol in `ListSelectorViewController`
- Minor changes in `TaxClassStore`, `Model`, `MockProduct`, `AuthenticatedState`

## Testing
**Case 1**
1) Navigate to a product detail with the tax class configured as `standard` (the tax class was never changed)
2) Tap Edit
3) Tap the cell price
4) Check that you are able to see the Tax Status cell configured, and the tax class as "Standard".
5) Try to change tax status and tax class, and when you press back, you are able to see the changes in the tax status and tax class cells.

**Case 2**
1) Delete and reinstall the app, or change store (with no tax classes already downloaded).
2) Navigate to a product detail with the tax class different from `standard`
3) Tap edit
4) Enable the network link conditioner
5) Tap the cell price
6) Check that you are able to see the tax class cell configured with the tax class slug, which will be updated with the tax class name when tax classes are fetched.
7) Play with tax status and tax class.

## Screenshots
| Edit Price screen             |  Tax Status picker |  Tax Class picker |
:-------------------------:|:-------------------------:|:-------------------------:
Simulator Screen Shot - iPhone 11 Pro - 2019-12-31 at 15 10 02 | Simulator Screen Shot - iPhone 11 Pro - 2019-12-31 at 15 10 04 | Simulator Screen Shot - iPhone 11 Pro - 2019-12-31 at 15 10 41

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/List Selector Data Source/ProductTaxClassListSelectorDataSource.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+struct ProductTaxClassListSelectorDataSource: PaginatedListSelectorDataSource {
+
+    typealias StorageModel = StorageTaxClass
+
+    var selected: TaxClass?
+
+    private let siteID: Int64
+
+    init(product: Product, selected: TaxClass?) {
+        self.siteID = Int64(product.siteID)
```
[the remark this answers is not in the document]
Comment (pmusolino) [reply WooCommerce/Classes/ViewRelated/Products/Edit Product/List Selector Data Source/ProductTaxClassListSelectorDataSource.swift]: I will align this branch with `develop` 😉

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/List Selector Data Source/ProductTaxClassListSelectorDataSource.swift [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                                                              sortedBy: [descriptor])
+    }
+
+    mutating func handleSelectedChange(selected: TaxClass) {
+        self.selected = selected
+    }
+
+    func configureCell(cell: BasicTableViewCell, model: TaxClass) {
+        cell.selectionStyle = .default
+        cell.isSelected = model.slug == selected?.slug
```
[the remark this answers is not in the document]
Comment (pmusolino) [reply WooCommerce/Classes/ViewRelated/Products/Edit Product/List Selector Data Source/ProductTaxClassListSelectorDataSource.swift]: It seems a good solution! 💯

**Comment by pmusolino** [comment]
@jaclync I updated the PR following your suggestions and aligning develop to this PR 😉 it's ready for a second pass!

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/List Selector Data Source/PaginatedProductShippingClassListSelectorDataSource.swift [hunk 3]
```diff
@@ -28,6 +28,10 @@ struct PaginatedProductShippingClassListSelectorDataSource: PaginatedListSelecto
         self.selected = selected == self.selected ? nil: selected
     }
 
+    func isSelected(model: ProductShippingClass) -> Bool {
+        return model == selected ? true : false
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Products/Edit Product/List Selector Data Source/PaginatedProductShippingClassListSelectorDataSource.swift]: nit suggestion: `return model == selected` might read a bit shorter :)

File: WooCommerce/Classes/ViewRelated/ListSelector/ListSelectorViewController.swift [hunk 4]
```diff
@@ -15,6 +15,9 @@ protocol ListSelectorDataSource {
     /// Called when a different model is selected.
     mutating func handleSelectedChange(selected: Model)
 
+    /// Configures the selected UI.
+    mutating func isSelected(model: Model) -> Bool
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/ListSelector/ListSelectorViewController.swift]: nit: just wondering, does the function need to be mutating?

File: WooCommerce/Classes/ViewRelated/ListSelector/PaginatedListSelectorViewController.swift [hunk 5]
```diff
@@ -17,6 +17,9 @@ protocol PaginatedListSelectorDataSource {
     /// Called when a different model is selected.
     mutating func handleSelectedChange(selected: StorageModel.ReadOnlyType)
 
+    /// Configures the selected UI.
+    mutating func isSelected(model: StorageModel.ReadOnlyType) -> Bool
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/ListSelector/PaginatedListSelectorViewController.swift]: nit: similar to above about `mutating` :)

**Comment by pmusolino** [comment]
Thanks for the review @jaclync! I updated it following your nits ;) 

Merging it now 👍

**Pull request merged**